esm_environment/esm_environment.py

```diff
@@ -164,10 +387,31 @@ def get_shell_commands(self):
         environment.append("")
         if "export_vars" in self.config:
             for var in self.config["export_vars"]:
-                if isinstance(var, dict):
-                    key = list(var.keys())[0]
-                    value = var[key]
-                    environment.append("export " + key + "='" + str(value) + "'")
+                # If export_vars is a dictionary
+                if isinstance(self.config["export_vars"], dict):
+                    # If the variable is a dictionary itself (e.g. "AWI_FESOM_YAML"
+                    # in fesom-1.4)
+                    if isinstance(self.config["export_vars"][var], dict):
+                        key = var
+                        value = self.config["export_vars"][key]
+                        environment.append("export " + key + "='" + str(value) + "'")
```

Comment:
Well done! 👏 
One point that I would like to make is that, at some point we need to adopt the best practices in Python and PEP recommendations. For example, the string concatenations are long and can sometimes be error prone. An alternative `f-string` version would be
`f"export  {key}='{value}'"`
I personally find these easier to read compared to the code given above. Additionally, by avoiding the `str()` function there can even be a slight performance gain.